netbox_onboarding/models.py

```diff
@@ -11,10 +11,23 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+from django.db.models.signals import post_save
+from django.dispatch import receiver
 from django.db import models
+from dcim.models import Device
 from .choices import OnboardingStatusChoices, OnboardingFailChoices
 
 
+@receiver(post_save, sender=Device)
+def init_onboarding_for_new_device(sender, instance, created, **kwargs):  # pylint: disable=W0613
+    """Register to create a OnboardingDevice object for each new Device Object using Django Signal.
+
+    https://docs.djangoproject.com/en/3.0/ref/signals/#post-save
+    """
+    if created:
+        OnboardingDevice.objects.create(device=instance)
```

Comment:
Per our discussion, I understand this now - you want an OnboardingDevice per Device so that we can control whether to re-onboard a device on a subsequent run by setting/checking the `OnboardingDevice.enabled` flag.